### PR TITLE
[Backport stable/8.1] ci(benchmark): add benchmark workflow

### PR DESCRIPTION
**Pull request by megglos**
## Description

Introduces a benchmark workflow local to the 8.1 stable branch. This allows us to make use of the [`workflow_dispatch`](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event) API to run benchmark workflows compatible with the GHA setup of a certain stable branch or patch release branch.

## Related issues

relates #13787

### CODE REVIEW

**Comment by megglos** [comment]
Successful benchmark run https://github.com/camunda/zeebe/actions/runs/5875832296/job/15932804075

**Comment by megglos** [comment]
need to check the measure job

**Comment by megglos** [comment]
@rodrigo-lourenco-lopes is this the measure scrip error you are looking into?
```
Waiting for minimal throughput
jq: error (at <stdin>:0): null (null) cannot be parsed as a number
Error: Process completed with exit code 5.
```
https://github.com/camunda/zeebe/actions/runs/5877547106/job/15938103975

**Comment by rodrigo-lourenco-lopes** [comment]
> @rodrigo-lourenco-lopes is this the measure scrip error you are looking into?
> 
> ```
> Waiting for minimal throughput
> jq: error (at <stdin>:0): null (null) cannot be parsed as a number
> Error: Process completed with exit code 5.
> ```
> 
> https://github.com/camunda/zeebe/actions/runs/5877547106/job/15938103975

I am looking into this issue, turns out that this issue is happening on 100% of the runs when we measure latency impact.
The changes that I had applied (and already reverted) did not solve the issue.

**Comment by megglos** [comment]
@oleschoenburg from wokflow setup this is ready to be merged, the failure on the measure step is preexisting

**Comment by Zelldon** [comment]
@megglos anything specific I look into, looks like the exact copy of https://github.com/camunda/zeebe/blob/main/.github/workflows/benchmark.yml

File: benchmarks/project/README.md [hunk 1]
```diff
@@ -4,16 +4,11 @@ This project contains code for the starter and worker, which are used during our
...
-
-### Build and push the images
-
-```
-docker-compose build
-docker-compose push
+```bash
+./mvnw -am -pl benchmarks/project package -DskipTests -DskipChecks
+./mvnw -pl benchmarks/project jib:build -Pstarter
+./mvnw -pl benchmarks/project jib:build -Pworker
```
Comment (Zelldon) [inline benchmarks/project/README.md:12]: 🚀

**Review by ChrisKujawa — APPROVED** [review]
Looks good from my perspective under the assumption that you have tested it :D

**Comment by megglos** [comment]
bors merge

**Pull request merged**